### PR TITLE
Example java

### PR DESCRIPTION
**Pull request by jburel**
Review the examples under training, modify reading of initial parameter so it can be part of CI job.
Marking on hold for now.

### CODE REVIEW

**Comment by sbesson** [comment]
One minor warning while compiling
```
[hudson@ome-c6100-3 java]$ javac -Xlint:unchecked src/training/util/*.java src/training/*.java
src/training/ReadDataAdvanced.java:223: warning: [unchecked] unchecked conversion
			Set<ImageData> images = dataset.getImages();
			                                         ^
  required: Set<ImageData>
  found:    Set
1 warning
```

**Comment by sbesson** [comment]
Another NPE reported by running `Setup`

```
java.lang.NullPointerException
	at training.ReadData.loadImage(ReadData.java:222)
	at training.ReadData.<init>(ReadData.java:353)
	at training.Setup.<init>(Setup.java:134)
	at training.Setup.main(Setup.java:147)
```

**Comment by sbesson** [comment]
To be on the safe side, the `omero.group` stored in `training_ice.config` could be parsed by the examples. At the moment we do not store any `plateAcquisitionid`. Either we need to store this in the setup or modify the training examples making use of it.

**Comment by jburel** [comment]
NPE due to the fact that there is not a plate acquisition. I will remove the example in this round so we do not have to modify the global config

**Comment by jburel** [comment]
@sbesson: 
You do not parse in matlab the groupId see ```parseOmeroProperties```

**Comment by sbesson** [comment]
@jburel: in the MATLAB examples, the `omero.client` is constructed directly using the the path to the Ice config file - see https://github.com/openmicroscopy/openmicroscopy/blob/v5.1.0/components/tools/OmeroM/src/connectOmero.m#L52. This handles the group logic without requiring `omero.group` to be explicitly parsed.

**Comment by jburel** [comment]
@sbesson: we should be good for tomorrow

File: examples/Training/java/src/training/Setup.java [hunk 1]
```diff
@@ -97,21 +97,27 @@ private String resolveFilePath(String fileName, String directory)
...
 			info.setHostName(p.getProperty("omero.host"));
 			info.setPassword(p.getProperty("omero.pass"));
 			info.setUserName(p.getProperty("omero.user"));
-			info.setImageId(Long.parseLong(p.getProperty("omero.imageId")));
-			info.setDatasetId(Long.parseLong(p.getProperty("omero.datasetId")));
-			info.setProjectId(Long.parseLong(p.getProperty("omero.projectId")));
-			info.setPlateId(Long.parseLong(p.getProperty("omero.plateId")));
-			info.setPlateAcquisitionId(Long.parseLong(
-					p.getProperty("omero.plateAcquisitionId")));
+			info.setGroupId(Long.parseLong(p.getProperty("omero.group")));
```
Comment (sbesson) [inline examples/Training/java/src/training/Setup.java]: See https://ci.openmicroscopy.org/job/OMERO-5.1-merge-training/293/console, `omero.group` is currently stored as a String
```
+ java -cp .:/opt/hudson/workspace/OMERO-5.1-merge-training/OMERO.matlab/libs/omero_client.jar:/opt/hudson/workspace/OMERO-5.1-merge-training/OMERO.matlab/libs/slf4j-api.jar training/Setup
java.lang.NumberFormatException: For input string: "training_group"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Long.parseLong(Long.java:441)
	at java.lang.Long.parseLong(Long.java:483)
	at training.Setup.<init>(Setup.java:114)
	at training.Setup.main(Setup.java:145)
```

Comment (jburel) [inline examples/Training/java/src/training/Setup.java]: ok, fixing that now

**Comment by sbesson** [comment]
Looking at the output of https://ci.openmicroscopy.org/job/OMERO-5.1-merge-training/294/console, examples are not passing. One NPE to fix but that can happen somewhere else. Merging.

**Comment by jburel** [comment]
NPE fixed in https://github.com/openmicroscopy/openmicroscopy/pull/3701

File: examples/Training/java/src/training/util/ReadOnlyByteArray.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
+/*
+ * org.openmicroscopy.shoola.util.mem.ReadOnlyByteArray
```
Comment (joshmoore) [inline examples/Training/java/src/training/util/ReadOnlyByteArray.java]: Might have been noted elsewhere, but this could likely use some cleaning up.

Comment (jburel) [inline examples/Training/java/src/training/util/ReadOnlyByteArray.java]: I can clean up in my follow-up PR.

Comment (jburel) [inline examples/Training/java/src/training/util/ReadOnlyByteArray.java]: Handled in https://github.com/openmicroscopy/openmicroscopy/pull/3701

**Pull request merged**